src/utils/timestampUtils.tsx

```diff
@@ -0,0 +1,81 @@
+import TimeAgo from "javascript-time-ago";
+import en from "javascript-time-ago/locale/en";
+import { ReduxMessage } from "./../store/reducers/messageListReducer";
+import * as _ from "lodash";
+
+const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+TimeAgo.addLocale(en);
+const timeAgo = new TimeAgo(userTimezone);
+
+function formatMessageTime(dateString: Date, lastMessage = false) {
+  const date = new Date(dateString);
+  const currentDate = new Date();
+
+  if (lastMessage && currentDate.toDateString() !== date.toDateString()) {
+    // If it's not the same day, show date.
+    return date.toDateString();
+  } else if (currentDate.getTime() - date.getTime() >= 3 * 60 * 60 * 1000) {
+    // If it's older than 3 hours, return AM/PM time
+    const hours = date.getHours();
+    const minutes = date.getMinutes();
+    const ampm = hours >= 12 ? "PM" : "AM";
```

Comment:
This locks timestamps to American date format, not everybody uses this.

Reply:
Good point. Let me look into this and come up with a proper solution.

Reply:
Hey @berkus 

I added a switch in the user profile modal that allows users to pick between 12 or 24 hour format timestamps. This preference is saved in local storage as well so that the user only has to set it once per browser. 

![Screen Recording 2023-11-02 at 11 12 49](https://github.com/twilio/twilio-conversations-demo-react/assets/51721338/f2ae2837-ead5-425a-b7ad-c25c3286c5d2)